jobclient/python/cook/scheduler/client/__init__.py

```diff
@@ -160,19 +157,20 @@ def submit(self, *,
 
         return UUID(uuid)
 
-    def query(self, uuid: UUID) -> Job:
+    def query(self, uuid: Union[str, UUID]) -> Job:
         """Query Cook for a job's status.
 
         :param uuid: The UUID to query.
-        :type uuid: UUID
+        :type uuid: Union[str, UUID]
         :return: A Job object containing the job's information.
         :rtype: Job
         """
+        uuid = str(uuid)
         if self.__job_endpoint == '/jobs':
             param_name = 'uuid'
         else:
             param_name = 'job'
```

Comment:
I think we should get rid of this complexity and only allow using `/jobs`, since the other endpoint is deprecated.